### DISCUSSION TITLE
wire:nagivate CSS tied to inline style violating CSP

### DISCUSSION THREAD

**Discussion by gofish543** [livewire/livewire#6459] [7. Bugs (archived)]
### Livewire version

3.0.0

### Which PHP version are you using?

PHP 8.2

### Repository URL

N/A

### Steps To Reproduce

1) Set a CSP policy for `style-src` **without** `unsafe-inline`
Something like the below code will work
```php
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\App;
use Illuminate\Support\Facades\Vite;
use Symfony\Component\HttpFoundation\Response;

class SecurityHeaders
{
    /**
     * Handle an incoming request.
     */
    public function handle(Request $request, Closure $next): Response
    {
        $nonce = Vite::useCspNonce();

        $response = $next($request);

        /** @link https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy */
        $response->headers->set('Content-Security-Policy', collect([
                "default-src 'self'",
                "script-src 'self' 'unsafe-eval' 'nonce-{$nonce}'", // Unsafe Eval for AlpineJS
                "style-src 'self' 'nonce-{$nonce}'",
        ])->implode(';'));

        return $response;
    }
}
```
2) Observe the response
![image](https://github.com/livewire/livewire/assets/10394554/415f6ad5-8844-42bf-9c64-719daff8412a)
![image](https://github.com/livewire/livewire/assets/10394554/29435d60-cc7e-452c-b290-0a79702d952b)

Can this be refactored into the Livewire styles which already support a nonce?

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by piotrku** [comment]
When using nonce, outputting livewire styles:
`@livewireStyles(['nonce' => csp_nonce()])`

instead of:
`@livewireStyles()`

should solve the issue.
